modin/pandas/dataframe.py

```diff
@@ -424,7 +423,16 @@ def groupby(
         # strings is passed in, the data used for the groupby is dropped before the
         # groupby takes place.
         drop = False
-
+        # Check that there is no ambiguity in the parameter we were given.
+        _by_check = by if is_list_like(by) else [by]
+        for k in _by_check:
+            if k in self.index.names and k in self.axes[axis]:
+                level_name, index_name = "an index", "a column"
+                if axis == 1:
+                    level_name, index_name = index_name, level_name
+                raise ValueError(
```

Comment:
nit: it's possible for multiple `k` to be invalid and cause us to raise a ValueError, however currently only one such `k` will be raised. Is it worth collecting all invalid `k` and raising a ValueError if any `k` are invalid?

Reply:
I checked, and in pandas, only the first invalid `k` is raised!